src/tests/calculate.test.js

```diff
@@ -0,0 +1,58 @@
+import calculate from '../logic/calculate';
+import '@testing-library/jest-dom';
+
+describe('Testing calculate logic function', () => {
+  test('Number should be enter first in calculate file', () => {
+    let obj = { total: null, next: null, operation: null };
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual(undefined);
+  });
+  test('Undefined if second number is not in calculate file', () => {
+    let obj = { total: null, next: null, operation: null };
+    obj = calculate(obj, '5');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual(undefined);
+  });
+  test('Plus function in calculate file', () => {
+    let obj = { total: null, next: null, operation: null };
+    obj = calculate(obj, '5');
+    obj = calculate(obj, '+');
+    obj = calculate(obj, '10');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual('15');
+  });
+  test('Minus function in calculate file', () => {
+    let obj = { total: null, next: null, operation: null };
+    obj = calculate(obj, '20');
+    obj = calculate(obj, '-');
+    obj = calculate(obj, '15');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual('5');
+  });
+  test('Divide function in calculate file', () => {
+    let obj = { total: null, next: null, operation: null };
+    obj = calculate(obj, '125');
+    obj = calculate(obj, '÷');
+    obj = calculate(obj, '5');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual('25');
+  });
+  test('Multiplication function in calculate file', () => {
+    let obj = { total: null, next: null, operation: null };
+    obj = calculate(obj, '30');
+    obj = calculate(obj, 'x');
+    obj = calculate(obj, '20');
+    obj = calculate(obj, '=');
+    expect(obj.total).toEqual('600');
+  });
+  test('Float calculation in calculate file', () => {
```

Comment:
Please add tests for `AC` and `+/-` buttons. All functions must be tested to avoid surprises when using the app.

Reply:
Well solved